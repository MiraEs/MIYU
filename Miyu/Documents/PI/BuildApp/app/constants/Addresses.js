'use strict';

export const US = 'US';
export const CANADA = 'CA';

export const SHIPPING_ADDRESS = 'SHIPPING';
export const BILLING_ADDRESS = 'BILLING';

export const COUNTRIES = [{
	text: 'United States',
	value: US,
}, {
	text: 'Canada',
	value: CANADA,
}];

export const STATES = [
	{text: 'Alabama', value: 'AL', country: US},
	{text: 'Alaska', value: 'AK', country: US},
	{text: 'American Samoa', value: 'AS', country: US},
	{text: 'Arizona', value: 'AZ', country: US},
	{text: 'Arkansas', value: 'AR', country: US},
	{text: 'California', value: 'CA', country: US},
	{text: 'Colorado', value: 'CO', country: US},
	{text: 'Connecticut', value: 'CT', country: US},
	{text: 'Delaware', value: 'DE', country: US},
	{text: 'District of Columbia', value: 'DC', country: US},
	{text: 'Federated States of Micronesia', value: 'FM', country: US},
	{text: 'Florida', value: 'FL', country: US},
	{text: 'Georgia', value: 'GA', country: US},
	{text: 'Guam', value: 'GU', country: US},
	{text: 'Hawaii', value: 'HI', country: US},
	{text: 'Idaho', value: 'ID', country: US},
	{text: 'Illinois', value: 'IL', country: US},
	{text: 'Indiana', value: 'IN', country: US},
	{text: 'Iowa', value: 'IA', country: US},
	{text: 'Kansas', value: 'KS', country: US},
	{text: 'Kentucky', value: 'KY', country: US},
	{text: 'Lousiana', value: 'LA', country: US},
	{text: 'Maine', value: 'ME', country: US},
	{text: 'Marshall Islands', value: 'MH', country: US},
	{text: 'Maryland', value: 'MD', country: US},
	{text: 'Massachusetts', value: 'MA', country: US},
	{text: 'Michigan', value: 'MI', country: US},
	{text: 'Minnesota', value: 'MN', country: US},
	{text: 'Mississippi', value: 'MS', country: US},
	{text: 'Missouri', value: 'MO', country: US},
	{text: 'Montana', value: 'MT', country: US},
	{text: 'Nebraska', value: 'NE', country: US},
	{text: 'Nevada', value: 'NV', country: US},
	{text: 'New Hampshire', value: 'NH', country: US},
	{text: 'New Jersey', value: 'NJ', country: US},
	{text: 'New Mexico', value: 'NM', country: US},
	{text: 'New York', value: 'NY', country: US},
	{text: 'North Carolina', value: 'NC', country: US},
	{text: 'North Dakota', value: 'ND', country: US},
	{text: 'Northern Mariana Islands', value: 'MP', country: US},
	{text: 'Ohio', value: 'OH', country: US},
	{text: 'Oklahoma', value: 'OK', country: US},
	{text: 'Oregon', value: 'OR', country: US},
	{text: 'Palau', value: 'PW', country: US},
	{text: 'Pennsylvania', value: 'PA', country: US},
	{text: 'Puerto Rico', value: 'PR', country: US},
	{text: 'Rhode Island', value: 'RI', country: US},
	{text: 'South Carolina', value: 'SC', country: US},
	{text: 'South Dakota', value: 'SD', country: US},
	{text: 'Tennessee', value: 'TN', country: US},
	{text: 'Texas', value: 'TX', country: US},
	{text: 'Utah', value: 'UT', country: US},
	{text: 'Vermont', value: 'VT', country: US},
	{text: 'Virgin Islands', value: 'VI', country: US},
	{text: 'Virginia', value: 'VA', country: US},
	{text: 'Washington', value: 'WA', country: US},
	{text: 'West Virginia', value: 'WV', country: US},
	{text: 'Wisconsin', value: 'WI', country: US},
	{text: 'Wyoming', value: 'WY', country: US},
	{text: 'Armed Forces Europe, the Middle East, and Canada', value: 'AE', country: US},
	{text: 'Armed Forces Canada', value: 'AE', country: US},
	{text: 'Armed Forces the Middle East', value: 'AE', country: US},
	{text: 'Armed Forces Europe', value: 'AE', country: US},
	{text: 'Armed Forces Pacific', value: 'AP', country: US},
	{text: 'Armed Forces Americas (Except Canada)', value: 'AA', country: US},
	{text: 'Armed Forces Americas', value: 'AA', country: US},
	{text: 'Alberta', value: 'AB', country: CANADA},
	{text: 'British Columbia', value: 'BC', country: CANADA},
	{text: 'Manitoba', value: 'MB', country: CANADA},
	{text: 'New Brunswick', value: 'NB', country: CANADA},
	{text: 'Newfoundland and Labrador', value: 'NL', country: CANADA},
	{text: 'Northwest Territories', value: 'NT', country: CANADA},
	{text: 'Nova Scotia', value: 'NS', country: CANADA},
	{text: 'Nunavut', value: 'NU', country: CANADA},
	{text: 'Ontario', value: 'ON', country: CANADA},
	{text: 'Prince Edward Island', value: 'PE', country: CANADA},
	{text: 'Quebec', value: 'QC', country: CANADA},
	{text: 'Saskatchewan', value: 'SK', country: CANADA},
	{text: 'Yukon', value: 'YT', country: CANADA},
];

export const STATE_EDIT_ADDRESS = 'EDIT_ADDRESS';
export const STATE_ADD_ADDRESS = 'ADD_ADDRESS';
export const STATE_CHECKOUT_ADDRESS = 'CHECKOUT_ADDRESS';
