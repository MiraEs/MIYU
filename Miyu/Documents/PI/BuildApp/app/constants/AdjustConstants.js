
const adjustTokens = {
	ADD_TO_CART: 'epirb3',
	CREATE_PROJECT: 'lpsglo',
	CALL_EXPERT: 'ly8kxo',
	CHECKOUT_COMPLETE: '9hgsev',
	CHECKOUT_START: 'o9elaa',
	CREATE_ACCOUNT: '1co7el',
	FAVORITE_ITEM: 'tm5k1k',
	FIRST_LAUNCH: 'mk4acg',
	INSTALL: 'mxz346',
	FIRST_TIME_LOGIN: '9f7nph',
};

module.exports = adjustTokens;
