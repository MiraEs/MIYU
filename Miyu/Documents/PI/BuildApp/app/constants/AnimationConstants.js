export const ANIMATION_TIMEOUT_200 = 200;
export const FADE_TEXT_DURATION = 400;
export const TIMEOUT_DURATION_2 = 2000;
export const TIMEOUT_DURATION_3 = 3000;
