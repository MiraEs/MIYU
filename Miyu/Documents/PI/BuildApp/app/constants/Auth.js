export const REQUEST_CREATE_USER = 'REQUEST_CREATE_USER';
export const UPDATE_USER = 'UPDATE_USER';
export const CREATE_USER_FAIL = 'CREATE_USER_FAIL';
export const UPDATE_USER_FAIL = 'UPDATE_USER_FAIL';

export const GET_CUSTOMER_ID_SUCCESS = 'GET_CUSTOMER_ID_SUCCESS';
export const GET_CUSTOMER_ID_FAIL = 'GET_CUSTOMER_ID_FAIL';

export const GET_CUSTOMER_FAIL = 'GET_CUSTOMER_FAIL';
export const GET_CUSTOMER_SUCCESS = 'GET_CUSTOMER_SUCCESS';

export const KEYCHAIN_FETCH_CREDS_FAIL = 'KEYCHAIN_FETCH_CREDS_FAIL';

export const LOGIN_USER_SUCCESS = 'LOGIN_USER_SUCCESS';
export const LOGIN_USER_ERROR = 'LOGIN_USER_ERROR';
export const LOGIN_USER_BEGIN = 'LOGIN_USER_BEGIN';
export const LOGIN_SOCIAL_USER_BEGIN = 'LOGIN_SOCIAL_USER_BEGIN';
export const LOGIN_USER_FINISH = 'LOGIN_USER_FINISH';
export const LOGIN_USER_RESET = 'LOGIN_USER_RESET';

export const PRE_AUTH_PROJECT_DATA = 'PRE_AUTH_PROJECT_DATA';

export const SIGN_USER_OUT = 'SIGN_USER_OUT';

export const AUTH_STATUS_UNKNOWN = 'AUTH_STATUS_UNKNOWN';
export const AUTH_STATUS_ANONYMOUS = 'AUTH_STATUS_ANONYMOUS';
export const AUTH_STATUS_AUTHENTICATED = 'AUTH_STATUS_AUTHENTICATED';

export const UPDATE_SIGNUP_ERROR = 'UPDATE_SIGNUP_ERROR';

export const UPDATE_DEVICE_TOKEN = 'UPDATE_DEVICE_TOKEN';

export const GET_CUSTOMER_REP_SUCCESS = 'GET_CUSTOMER_REP_SUCCESS';
export const GET_CUSTOMER_REP_FAIL = 'GET_CUSTOMER_REP_FAIL';

export const GET_CUSTOMER_REWARDS_TIER = 'GET_CUSTOMER_REWARDS_TIER';

export const SET_IMPERSONATOR_ID = 'SET_IMPERSONATOR_ID';
export const HAS_LOGGED_IN_ON_DEVICE = 'HAS_LOGGED_IN_ON_DEVICE';
