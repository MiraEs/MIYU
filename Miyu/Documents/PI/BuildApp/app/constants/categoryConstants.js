'use strict';

const categoryConstants = {

	CATEGORIES_SUCCESS: 'CATEGORIES_SUCCESS',
	CATEGORY_SUCCESS: 'CATEGORY_SUCCESS',
	CATEGORIES_FAIL: 'CATEGORIES_FAIL',

};

module.exports = categoryConstants;
