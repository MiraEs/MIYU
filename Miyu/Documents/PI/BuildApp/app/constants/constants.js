'use strict';

const constants = {
	SITE_ID: '82',
	STORE_ID: '248',

	BANNER_VISIBLE_TIMEOUT: 3000,
	BANNER_SLIDE_DOWN: 700,
	BANNER_SLIDE_UP: 800,

	GET_CUSTOMER_ID_SUCCESS: 'GET_CUSTOMER_ID_SUCCESS',
	GET_CUSTOMER_ID_FAIL: 'GET_CUSTOMER_ID_FAIL',

	KEYCHAIN_FETCH_CREDS_FAIL: 'KEYCHAIN_FETCH_CREDS_FAIL',

	LOGIN_USER_ERROR: 'LOGIN_USER_ERROR',

	LOAD_EVENTS: 'LOAD_EVENTS',
	GLOBAL_LOAD_EVENTS_SUCCESS: 'GLOBAL_LOAD_EVENTS_SUCCESS',
	GLOBAL_LOAD_EVENTS_ERROR: 'GLOBAL_LOAD_EVENTS_ERROR',

	GLOBAL_LOAD_NEXT_EVENTS_SUCCESS: 'GLOBAL_LOAD_NEXT_EVENTS_SUCCESS',
	GLOBAL_LOAD_NEXT_EVENTS_FAIL: 'GLOBAL_LOAD_NEXT_EVENTS_FAIL',

	GLOBAL_ADD_EVENT_SUCCESS: 'GLOBAL_ADD_EVENT_SUCCESS',
	GLOBAL_ADD_EVENT_ERROR: 'GLOBAL_ADD_EVENT_ERROR',

	GLOBAL_SAVE_COMMENT: 'GLOBAL_SAVE_COMMENT',
	GLOBAL_SAVE_COMMENT_SUCCESS: 'GLOBAL_SAVE_COMMENT_SUCCESS',
	GLOBAL_SAVE_COMMENT_ERROR: 'GLOBAL_SAVE_COMMENT_ERROR',

	PROJECT_LOAD_EVENTS_SUCCESS: 'PROJECT_LOAD_EVENTS_SUCCESS',
	PROJECT_LOAD_EVENTS_ERROR: 'PROJECT_LOAD_EVENTS_ERROR',

	PROJECT_LOAD_NEXT_EVENTS_SUCCESS: 'PROJECT_LOAD_NEXT_EVENTS_SUCCESS',
	PROJECT_LOAD_NEXT_EVENTS_FAIL: 'PROJECT_LOAD_NEXT_EVENTS_FAIL',

	PROJECT_ADD_EVENT: 'PROJECT_ADD_EVENT',
	PROJECT_ADD_EVENT_SUCCESS: 'PROJECT_ADD_EVENT_SUCCESS',
	PROJECT_ADD_EVENT_ERROR: 'PROJECT_ADD_EVENT_ERROR',

	PROJECT_SAVE_COMMENT: 'PROJECT_SAVE_COMMENT',
	PROJECT_SAVE_COMMENT_SUCCESS: 'PROJECT_SAVE_COMMENT_SUCCESS',
	PROJECT_SAVE_COMMENT_ERROR: 'PROJECT_SAVE_COMMENT_ERROR',

	LOAD_PHOTOS_SUCCESS: 'LOAD_PHOTOS_SUCCESS',
	LOAD_PHOTOS_FAIL: 'LOAD_PHOTOS_FAIL',
	LOADING_PHOTOS: 'LOADING_PHOTOS',

	SEND_PROJECT_INVITES: 'SEND_PROJECT_INVITES',
	SEND_PROJECT_INVITES_SUCCESS: 'SEND_PROJECT_INVITES_SUCCESS',
	SEND_PROJECT_INVITES_ERROR: 'SEND_PROJECT_INVITES_ERROR',

	LOAD_TEAM_SUCCESS: 'LOAD_TEAM_SUCCESS',
	LOAD_TEAM_FAIL: 'LOAD_TEAM_FAIL',

	LOAD_INVITEES_SUCCESS: 'LOAD_INVITEES_SUCCESS',
	LOAD_INVITEES_FAIL: 'LOAD_INVITEES_FAIL',

	RESEND_INVITE_BEGIN: 'RESEND_INVITE_BEGIN',
	RESEND_INVITE: 'RESEND_INVITE',
	RESEND_INVITE_SUCCESS: 'RESEND_INVITE_SUCCESS',
	RESEND_INVITE_FAIL: 'RESEND_INVITE_FAIL',
	RESEND_ALL_INVITES_SUCCESS: 'RESEND_ALL_INVITES_SUCCESS',
	RESEND_ALL_INVITES_FAIL: 'RESEND_ALL_INVITES_FAIL',
	IS_RESENDING_INVITES: 'IS_RESENDING_INVITES',

	SAVE_PROJECT_SUCCESS: 'SAVE_PROJECT_SUCCESS',
	SAVE_PROJECT_FAIL: 'SAVE_PROJECT_FAIL',
	UPDATE_PROJECT_SUCCESS: 'UPDATE_PROJECT_SUCCESS',

	CHANGE_PROJECT_STATUS_SUCCESS: 'CHANGE_PROJECT_STATUS_SUCCESS',
	CHANGE_PROJECT_STATUS_FAIL: 'CHANGE_PROJECT_STATUS_FAIL',

	RECEIVE_PROJECTS_SUCCESS: 'RECEIVE_PROJECTS_SUCCESS',
	RECEIVE_PROJECTS_FAIL: 'RECEIVE_PROJECTS_FAIL',
	RECEIVE_SHOPPING_LISTS_SUCCESS: 'RECEIVE_SHOPPING_LISTS_SUCCESS',
	RECEIVE_SHOPPING_LISTS_FOR_PROJECT_SUCCESS: 'RECEIVE_SHOPPING_LISTS_FOR_PROJECT_SUCCESS',
	RECEIVE_SHOPPING_LIST_AGGREGATE_DATA: 'RECEIVE_SHOPPING_LIST_AGGREGATE_DATA',

	REJECT_PROJECT_INVITE: 'REJECT_PROJECT_INVITE',
	REJECT_PROJECT_INVITE_SUCCESS: 'REJECT_PROJECT_INVITE_SUCCESS',
	REJECT_PROJECT_INVITE_FAIL: 'REJECT_PROJECT_INVITE_FAIL',

	DELETE_PROJECT_TEAM_MEMBER: 'DELETE_PROJECT_TEAM_MEMBER',
	DELETE_PROJECT_TEAM_MEMBER_SUCCESS: 'DELETE_PROJECT_TEAM_MEMBER_SUCCESS',
	DELETE_PROJECT_TEAM_MEMBER_FAIL: 'DELETE_PROJECT_TEAM_MEMBER_FAIL',

	GET_CUSTOMER_SUCCESS: 'GET_CUSTOMER_SUCCESS',
	GET_CUSTOMER_FAIL: 'GET_CUSTOMER_FAIL',

	LOAD_SINGLE_EVENT_SUCCESS: 'LOAD_SINGLE_EVENT_SUCCESS',
	LOAD_SINGLE_EVENT_FAIL: 'LOAD_SINGLE_EVENT_FAIL',

	SINGLE_EVENT_SAVE_COMMENT: 'SINGLE_EVENT_SAVE_COMMENT',
	SINGLE_EVENT_SAVE_COMMENT_SUCCESS: 'SINGLE_EVENT_SAVE_COMMENT_SUCCESS',
	SINGLE_EVENT_SAVE_COMMENT_FAIL: 'SINGLE_EVENT_SAVE_COMMENT_FAIL',

	LOAD_DEVICE_PHOTOS_SUCCESS: 'LOAD_DEVICE_PHOTOS_SUCCESS',
	LOAD_DEVICE_PHOTOS_FAIL: 'LOAD_DEVICE_PHOTOS_FAIL',

	GET_NOTIFICATION_SETTINGS_SUCCESS: 'GET_NOTIFICATION_SETTINGS_SUCCESS',
	GET_NOTIFICATION_SETTINGS_FAIL: 'GET_NOTIFICATION_SETTINGS_FAIL',

	SAVE_NOTIFICATION_SETTINGS: 'SAVE_NOTIFICATION_SETTINGS',
	SAVE_NOTIFICATION_SETTINGS_SUCCESS: 'SAVE_NOTIFICATION_SETTINGS_SUCCESS',
	SAVE_NOTIFICATION_SETTINGS_FAIL: 'SAVE_NOTIFICATION_SETTINGS_FAIL',

	SOCIAL_LOGIN_PROMPT_CREDS: 'SOCIAL_LOGIN_PROMPT_CREDS',
	SOCIAL_GET_CLIENT_DATA: 'SOCIAL_GET_CLIENT_DATA',
	SOCIAL_LINK_USER: 'SOCIAL_LINK_USER',
	SOCIAL_LINK_USER_ERROR: 'SOCIAL_LINK_USER_ERROR',
	DID_LOGIN_WITH_SOCIAL: 'DID_LOGIN_WITH_SOCIAL',
	DID_OPEN_PREVIOUSLY: 'DID_OPEN_PREVIOUSLY',
	SOCIAL_LOGIN_TYPE: 'SOCIAL_LOGIN_TYPE',
	TOGGLE_PHOTO_SELECTED: 'TOGGLE_PHOTO_SELECTED',

	UPDATE_SIGNUP_ERROR: 'UPDATE_SIGNUP_ERROR',

	SHOULD_PRESENT_NO_EMAIL_ERROR_MESSAGE: 'SHOULD_PRESENT_NO_EMAIL_ERROR_MESSAGE',

	// data manipulation status
	DELETE_PENDING: 'DELETE_PENDING',
	DELETE_FAIL: 'DELETE_FAIL',

	ADD_PENDING: 'ADD_PENDING',
	ADD_FAIL: 'ADD_FAIL',

	UPDATE_PENDING: 'UPDATE_PENDING',
	UPDATE_FAIL: 'UPDATE_FAIL',

	SHOULD_SHOW_INTRO: 'SHOULD_SHOW_INTRO',

	FEED_PAGE_LENGTH: 20,

	PRE_AUTH_PROJECT_DATA: 'PRE_AUTH_PROJECT_DATA',

	ADDED: 'ADDED',

	DEMO_LOAD_EVENTS_SUCCESS: 'DEMO_LOAD_EVENTS_SUCCESS',

	FILTER_PHOTOS_SELECTED: 'FILTER_PHOTOS_SELECTED',

	REPLACE_SELECTED_PHOTOS: 'REPLACE_SELECTED_PHOTOS',

	ADD_PHOTO: 'ADD_PHOTO',

	TOGGLE_FETCHING_PHOTOS: 'TOGGLE_FETCHING_PHOTOS',

	LOAD_ORDERS: 'LOAD_ORDERS',
	LOAD_ORDERS_SUCCESS: 'LOAD_ORDERS_SUCCESS',
	LOAD_ORDERS_FAIL: 'LOAD_ORDERS_FAIL',

	LOAD_ORDER_DETAILS: 'LOAD_ORDER_DETAILS',
	LOAD_ORDER_DETAILS_SUCCESS: 'LOAD_ORDER_DETAILS_SUCCESS',
	LOAD_ORDER_DETAILS_FAIL: 'LOAD_ORDER_DETAILS_FAIL',

	LOAD_SHIPPING_INFO: 'LOAD_SHIPPING_INFO',
	LOAD_SHIPPING_INFO_SUCCESS: 'LOAD_SHIPPING_INFO_SUCCESS',
	LOAD_SHIPPING_INFO_FAIL: 'LOAD_SHIPPING_INFO_FAIL',

	LOAD_RETURNS: 'LOAD_RETURNS',
	LOAD_RETURNS_SUCCESS: 'LOAD_RETURNS_SUCCESS',
	LOAD_RETURNS_FAIL: 'LOAD_RETURNS_FAIL',

	LOAD_RETURN_DETAILS: 'LOAD_RETURN_DETAILS',
	LOAD_RETURN_DETAILS_SUCCESS: 'LOAD_RETURN_DETAILS_SUCCESS',
	LOAD_RETURN_DETAILS_FAIL: 'LOAD_RETURN_DETAILS_FAIL',

	// Returns reducer
	ORDER_TO_RETURN: 'ORDER_TO_RETURN',
	UPDATE_ITEM_IN_RETURN: 'UPDATE_ITEM_IN_RETURN',
	UPDATE_SHIPPING_METHODS: 'UPDATE_SHIPPING_METHODS',

	actions: {
		getProjects: 'get-projects',
		getEvents: 'get-events',
		signUserOut: 'sign-user-out',
	},

	user: {
		status: {
			unknown: 'unknown',
			anonymous: 'anonymous',
			authenticated: 'authenticated',
		},
	},
	EXPERT: '@BuildExpert',
	CLOUDINARY_URL: 'res.cloudinary.com',

	TERMS_OF_USE_URL: 'https://www.build.com/support/terms',
	SECURITY_AND_PRIVACY_URL: 'https://www.build.com/support/security',
	RETURNS_POLICY_URL: 'https://www.build.com/support/returns',

	ADD_PROJECT_ORDER_SUCCESS: 'ADD_PROJECT_ORDER_SUCCESS',

	ADD_PROJECT_ORDER_ERROR: 'ADD_PROJECT_ORDER_ERROR',

	ADD_PROJECT_ORDER: 'ADD_PROJECT_ORDER',

	TEMPLATE_ORDER: 'TEMPLATE_ORDER',
	TEMPLATE_SHIPPING_ESTIMATE: 'TEMPLATE_SHIPPING_ESTIMATE',

	GALLERY_GO_TO_INDEX: 'GALLERY_GO_TO_INDEX',

	TAP_RADIUS: 20,

	DOUBLE_TAP_DELAY: 300,

	// HalfPageListSelector heights
	HALF_HEIGHT: 'HALF',
	THREE_QUARTER_HEIGHT: 'THREE_QUARTERS',

	// constants for the tab bar
	MAIN: 'main',
	HOME: 'home',
	CART: 'cart',
	MORE: 'more',
	EXPERTS: 'experts',
	PROJECTS: 'projects',
	PROJECTSV2: 'projectsV2',
	LISTS: 'lists',

	// tab navigation for Checkout Screen
	ACTIVE: 'active',
	ENABLED: 'enabled',
	DISABLED: 'disabled',

	DEFAULT: 'default',

	FADE: 'fade',

	// Feature switches
	SET_FEATURE_STATE: 'SET_FEATURE_STATE',
	SET_FEATURE_WITH_EXPIRATION: 'SET_FEATURE_WITH_EXPIRATION',

	REFRESH_CURRENT_SCREEN: 'REFRESH_CURRENT_SCREEN',
	UNAUTHORIZED_ERROR: 'UNAUTHORIZED_ERROR',

	APP_LAUNCH_INFO: 'APP_LAUNCH_INFO',
	STORE_CUSTOMER_INFO: 'STORE_CUSTOMER_INFO',
	USER_LOGIN_STATUS: 'USER_LOGIN_STATUS',

	HIT_SLOP: {
		top: 14,
		left: 14,
		right: 14,
		bottom: 14,
	},

	STORE_NOTIFICATIONS_OPT_IN: 'STORE_NOTIFICATIONS_OPT_IN',
};

// DEV only constants
if (__DEV__) {
	constants.IMPERSONATOR_PASSWORD = 'IMPERSONATOR_PASSWORD';
}

module.exports = constants;
