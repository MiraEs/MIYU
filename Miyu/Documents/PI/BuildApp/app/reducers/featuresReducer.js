'use strict';

import {
	SET_FEATURE_STATE,
	SET_FEATURE_WITH_EXPIRATION,
} from '../constants/constants';
import { handleActions } from 'redux-actions';
import { REHYDRATE } from 'redux-persist/constants';

const initialState = {
	features: {
		facebookLoginIOS: true,
		facebookLoginAndroid: true,
		paypal: true,
		applePay: true,
		projects: false,
		typeAhead: true,
		geCheckout: true,
		onboarding: true,
		recentSearches: true,
		useBranchLinksForShare: true,
		contentHomeScreen: true,
		lookback: {
			enabled: false,
			default: false,
			expireAfter: null,
		},
		shoppingLists: true,
		actualTaxes: true,
		returns: false,
		whyDidYouUpdate: false,
	},
};

export function expireFeatures(features) {

	Object.keys(features || {}).forEach((key) => {
		const feature = features[key];
		const featureIsObject = feature && feature.hasOwnProperty('default');
		const featureHasExpireTime = feature && feature.expireAfter && typeof feature.expireAfter.getTime === 'function';
		const featureIsExpired = featureHasExpireTime && feature.expireAfter.getTime() < Date.now();
		if (featureIsObject && featureIsExpired) {
			features[key] = {
				enabled: feature.default,
				default: feature.default,
				expireAfter: null,
			};
		}
	});

	return features;

}

const featuresReducer = handleActions({
	[SET_FEATURE_STATE]: (state, action) => {
		return {
			features: {
				...state.features,
				[action.feature]: action.enabled,
			},
		};
	},
	[SET_FEATURE_WITH_EXPIRATION]: (state, action) => {
		return {
			...state,
			features: {
				...state.features,
				...action.payload,
			},
		};
	},
	[REHYDRATE]: (state, action) => {
		const { featuresReducer } = action.payload;
		if (featuresReducer) {
			return {
				...state,
				...featuresReducer,
				features: {
					...state.features,
					...featuresReducer.features,
					...expireFeatures(featuresReducer.features),
				},
			};
		}
		return state;
	},
}, initialState);

export default featuresReducer;
