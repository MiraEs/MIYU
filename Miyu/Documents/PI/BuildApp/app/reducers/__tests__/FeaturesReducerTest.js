import featuresReducer from '../featuresReducer';
import {
	SET_FEATURE_STATE,
	SET_FEATURE_WITH_EXPIRATION,
} from '../../constants/constants';
import { REHYDRATE } from 'redux-persist/constants';

const previousState = {
	features: {
		feature1: false,
		feature2: true,
	},
};

// This test will break 100 years from now.
// If this code base and function still exists then...
// Well, that will be someone else's problem.
const futureDate = new Date(2117, 3, 21);

describe('featuresReducer', () => {

	it('should return initialState', () => {
		expect(featuresReducer(undefined, {})).toMatchSnapshot();
	});

	it('should return the correct state for SET_FEATURE_STATE to true', () => {
		expect(featuresReducer(previousState, {
			type: SET_FEATURE_STATE,
			feature: 'feature1',
			enabled: true,
		})).toMatchSnapshot();
	});

	it('should return the correct state for SET_FEATURE_STATE to false', () => {
		expect(featuresReducer(previousState, {
			type: SET_FEATURE_STATE,
			feature: 'feature2',
			enabled: false,
		})).toMatchSnapshot();
	});

	it('should expire features if needed', () => {
		const reducer = {
			...previousState,
			features: {
				...previousState.features,
				expiredFeature: {
					enabled: true,
					default: false,
					expireAfter: new Date('2017-03-21T16:41:46.275Z'),
				},
				notExpiredFeature: {
					enabled: true,
					default: false,
					expireAfter: futureDate,
				},
			},
		};
		expect(featuresReducer(reducer, {
			type: REHYDRATE,
			payload: {
				featuresReducer: reducer,
			},
		})).toMatchSnapshot();
	});

	it('should return state if no features reducer is found', () => {
		expect(featuresReducer({}, {
			type: REHYDRATE,
			payload: {},
		})).toMatchSnapshot();
	});

	it('should set a new feature with expiration', () => {
		expect(featuresReducer(previousState, {
			type: SET_FEATURE_WITH_EXPIRATION,
			payload: {
				testFeature: {
					enabled: true,
					default: false,
					expireAfter: futureDate,
				},
			},
		})).toMatchSnapshot();
	});

});
