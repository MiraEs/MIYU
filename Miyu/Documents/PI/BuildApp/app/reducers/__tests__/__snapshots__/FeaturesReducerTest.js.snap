// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`featuresReducer should expire features if needed 1`] = `
Object {
  "features": Object {
    "expiredFeature": Object {
      "default": false,
      "enabled": false,
      "expireAfter": null,
    },
    "feature1": false,
    "feature2": true,
    "notExpiredFeature": Object {
      "default": false,
      "enabled": true,
      "expireAfter": 2117-04-21T07:00:00.000Z,
    },
  },
}
`;

exports[`featuresReducer should return initialState 1`] = `
Object {
  "features": Object {
    "actualTaxes": true,
    "applePay": true,
    "contentHomeScreen": true,
    "facebookLoginAndroid": true,
    "facebookLoginIOS": true,
    "geCheckout": true,
    "lookback": Object {
      "default": false,
      "enabled": false,
      "expireAfter": null,
    },
    "onboarding": true,
    "paypal": true,
    "projects": false,
    "recentSearches": true,
    "returns": false,
    "shoppingLists": true,
    "typeAhead": true,
    "useBranchLinksForShare": true,
    "whyDidYouUpdate": false,
  },
}
`;

exports[`featuresReducer should return state if no features reducer is found 1`] = `Object {}`;

exports[`featuresReducer should return the correct state for SET_FEATURE_STATE to false 1`] = `
Object {
  "features": Object {
    "feature1": false,
    "feature2": false,
  },
}
`;

exports[`featuresReducer should return the correct state for SET_FEATURE_STATE to true 1`] = `
Object {
  "features": Object {
    "feature1": true,
    "feature2": true,
  },
}
`;

exports[`featuresReducer should set a new feature with expiration 1`] = `
Object {
  "features": Object {
    "feature1": false,
    "feature2": true,
    "testFeature": Object {
      "default": false,
      "enabled": true,
      "expireAfter": 2117-04-21T07:00:00.000Z,
    },
  },
}
`;
